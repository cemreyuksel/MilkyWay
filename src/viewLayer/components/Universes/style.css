.star-items {
  display: grid;
  margin: 2rem 0 3rem 0;
  justify-items: center;
  grid-gap: 2rem;
}
