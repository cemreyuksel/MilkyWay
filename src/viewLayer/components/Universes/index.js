import React from 'react';

import { Card } from 'react-bootstrap';

import { useFetch } from '../../hooks';
import { UNIVERSE_URL, LOADING_MESSAGE } from '../../../dataLayer/env';

import './style.scss';

const Universes = () => {
  const [data, loading] = useFetch(UNIVERSE_URL);

  return (
    <>
      {loading ? (
        <h2>{`${LOADING_MESSAGE}`}</h2>
      ) : (
        <div className="universe-items">
          {data.map(({ maxSize, name }, index) => (
            <div key={index}>
              <Card
                bg={'Info'}
                style={{ width: '20rem' }}
                className="text-center"
              >
                <Card.Body>
                  <Card.Title>{name}</Card.Title>
                  <Card.Header>Max Size: {maxSize}</Card.Header>
                </Card.Body>
              </Card>
            </div>
          ))}
        </div>
      )}
    </>
  );
};

export default Universes;
