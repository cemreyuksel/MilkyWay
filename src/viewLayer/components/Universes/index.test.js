import React from 'react';
import TestRenderer from 'react-test-renderer';

import Universes from './index';

jest.mock('../../hooks', () => ({
  useFetch: () => [
    [
      {
        id: 333,
        maxSize: 3,
        name: 'Big universe',
      }
    ],
    false,
  ],
}));

describe('Universes component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('renders', () => {
    const tree = TestRenderer.create(<Universes />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test('displays the maximum and current size of the universe', () => {
    const expectedHeader = ["Max Size: ", "3"];
    const expectedTitle = ['Big universe'];
    const renderer = TestRenderer.create(
      <Universes />
    );
    const instance = renderer.root;
    expect(instance.findByProps({className: "card-header"}).children).toEqual(expectedHeader);
    expect(instance.findByProps({className: "card-title h5"}).children).toEqual(expectedTitle);
  });
});
