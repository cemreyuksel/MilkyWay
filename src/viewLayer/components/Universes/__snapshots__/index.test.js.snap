// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Universes component renders 1`] = `
<div
  className="universe-items"
>
  <div>
    <div
      className="text-center card bg-Info"
      style={
        Object {
          "width": "20rem",
        }
      }
    >
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
        >
          Big universe
        </div>
        <div
          className="card-header"
        >
          Max Size: 
          3
        </div>
      </div>
    </div>
  </div>
</div>
`;
